src/wait-for.js

```diff
@@ -101,9 +110,15 @@ function waitFor(
         reject(e)
         return
       }
-      intervalId = setInterval(checkRealTimersCallback, interval)
```

Comment:
Let me know if I have this right: We're still calling this every other time but now by scheduling a timeout every time `callback` fails?

So we're now somewhat drifting over time i.e. we're not checking every `interval` but rather at most every `interval` which is `interval + runtime(callback)`?

Reply:
Yes that's correct. This will avoid `setInterval` scheduling tasks while the `callback` is still running, which leads to `callback` running constantly and not relinquishing the main thread for other async tasks to complete.